default_gateway.go

```diff
@@ -75,7 +76,7 @@ func (sb *sandbox) clearDefaultGW() error {
 		return nil
 	}
 
-	if ep == sb.getGatewayEndpoint() && !sb.inDelete {
+	if ep == sb.getGatewayEndpoint() && !sb.inDelete && len(sb.getConnectedEndpoints()) != 1 {
```

Comment:
I think just the count of connected EPs may not be sufficient in all cases. For ex: if a container is first connected to an internal network and then an overlay, later when its disconnected from the overlay network shouldn't we disconnect it from the default gw network as well ?